x-pack/plugins/ml/public/application/components/custom_urls/custom_url_editor/utils.ts

```diff
@@ -42,25 +89,35 @@ export function getNewCustomUrlDefaults(job, dashboards, dataViews) {
 
   // For the Discover option, set the default data view to that
   // which matches the indices configured in the job datafeed.
-  const datafeedConfig = job.datafeed_config;
+  let query: estypes.QueryDslQueryContainer;
+  let indicesName: string | undefined;
+  let jobId;
   if (
+    isAnomalyDetectionJob(job) &&
     dataViews !== undefined &&
     dataViews.length > 0 &&
-    datafeedConfig !== undefined &&
-    datafeedConfig.indices !== undefined &&
-    datafeedConfig.indices.length > 0
+    job.datafeed_config !== undefined &&
+    job.datafeed_config.indices !== undefined &&
+    job.datafeed_config.indices.length > 0
   ) {
-    const indicesName = datafeedConfig.indices.join();
-    const defaultDataViewId = dataViews.find((dv) => dv.title === indicesName)?.id;
-    kibanaSettings.discoverIndexPatternId = defaultDataViewId;
-    kibanaSettings.filters =
-      defaultDataViewId === null
-        ? []
-        : getFiltersForDSLQuery(job.datafeed_config.query, defaultDataViewId, job.job_id);
+    indicesName = job.datafeed_config.indices.join();
+    query = job.datafeed_config?.query ?? {};
+    jobId = job.job_id;
+  } else if (isDataFrameAnalyticsConfigs(job) && dataViews !== undefined && dataViews.length > 0) {
+    indicesName = job.dest.index;
+    query = job.source?.query ?? {};
+    jobId = job.id;
   }
 
+  const defaultDataViewId = dataViews.find((dv) => dv.title === indicesName)?.id;
+  kibanaSettings.discoverIndexPatternId = defaultDataViewId;
+  kibanaSettings.filters =
+    // @ts-ignore
```

Comment:
why do we need this `@ts-ignore`?

Reply:
Removed in 642ce6dbe70bb719b6416e5a61184a6b966b71d7